### PR TITLE
[SQL] Add two compiler flags: --outputsAreSets and --ignoreOrder

### PR DESCRIPTION
**Pull request by mihaibudiu**
This PR addresses 3 problems:
- provides a flag to ignore ORDER BY statements that are last in a view definition (but not if followed by LIMIT)
- provides a compiler flag to ensure that all outputs are sets (not bags) 
- it implements a suboptimal version of ORDER BY with LIMIT. This should be revisited, but at least it allows us to run tests.

Is this a user-visible change (yes/no): yes

This is the compiler part of #871

### CODE REVIEW

**Comment by mihaibudiu** [comment]
When trying to use the pre-push script I got lots of rustdoc errors, so I fixed these as well as I could

**Comment by gz** [comment]
I added the doc command to CI in your branch.

**Comment by gz** [comment]
@mihaibudiu something from your compiler?

```
        +test-python *failed* | dbsp.error.CompilationException: Unexpected project status: ProgramStatusType7(system_error='Only one main parameter allowed but found several: "/working-dir/cargo_workspace/project018b2ba1-66b8-7466-a1b1-121bf94a0fa1/project.sql" and "-alltables"\n\nexit code: exit status: 1', additional_properties={})
```

**Comment by mihaibudiu** [comment]
The alltables flag now requires two dashes

**Review by lalithsuresh — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Two new cmopiler flags: `--outputAreSets` and `--ignoreOrder`
```
Comment (lalithsuresh) [inline CHANGELOG.md:17]: compiler*

**Comment by ryzhyk** [comment]
> @mihaibudiu something from your compiler?
> 
> ```
>         +test-python *failed* | dbsp.error.CompilationException: Unexpected project status: ProgramStatusType7(system_error='Only one main parameter allowed but found several: "/working-dir/cargo_workspace/project018b2ba1-66b8-7466-a1b1-121bf94a0fa1/project.sql" and "-alltables"\n\nexit code: exit status: 1', additional_properties={})
> ```

@gz, this is probably unrelated to this PR, I will look into it.

**Review by ryzhyk — COMMENTED** [review]
Should it be `--outputsAreSets` ?

**Pull request merged**